### ISSUE TITLE
Servers list

### ISSUE THREAD

**Issue by Stun3** [veso266/Flash-Speedtest.net-by-Ookla#4]
Hey guys, I was wondering if its possible to retrieve a list of new servers in .xml file format for my website (https://stun3.github.io/speedtest/)
Thanks

**Comment by veso266** [comment]
hmm, this are the only new servers left: https://www.speedtest.net/speedtest-servers-static.php

the new servers the new speedtest is using are here: https://www.speedtest.net/api/js/servers?engine=js&limit=10&https_functional=true (you would need to convert them from json to xml)

not sure if they still work (if server protocol didnt change they should)

**Comment by Stun3** [comment]
Thanks, but for some reason my speedtest hosted on github pages ignores the server list and only shows some servers.
Do you have discord?

**Comment by veso266** [comment]
We can talk here, thats why issues are for. This also means that you and I can read eachothers messages when we have time and also means others that might have the same problem can see a solution

Hmm thats odd

Did u check with network inspector where the serverlist is pulled from? If it only shows some servers its possible its still pulling them from: speedtest.net not ur servers.xml

**Comment by Stun3** [comment]
Nope, sorry I don't know how to use a network inspector.
Thinking about that, it's possible its pulling from the speedtest.net not from  my servers.xml.

**Comment by Stun3** [comment]
Nevermind, I fixed it by copying the server.xml file contents into these 2 files:
speedtest-servers-static.php
speedtest-servers.php
and also by removing the line 
<?xml version="1.0" encoding="UTF-8"?>
from the beginning of the XML file before copying it

**Comment by Stun3** [comment]
I still have one problem tho, when i try to do a test on a WORKING server (its from the same provider and location in the new speedtest) it returns with a Latency Test Error. maybe you know why?
Thanks

**Comment by filmdestruction** [comment]
Hey stun3, looks like your website is working now. Thank you both veso266 for providing the code, and thank you stun3 for having a website up
![goodwebsite](https://github.com/veso266/Flash-Speedtest.net-by-Ookla/assets/54926371/0e2f98ff-b60d-4dbc-956d-6a1008fa57d4)

**Comment by Stun3** [comment]
Yeah, I made the whole website unlike some other guys here only making a index.html with the flash player, but  thanks veso266 for most of the code!

**Comment by Stun3** [comment]
Also, Report any dead servers to me
include: server name and country
the host (the thing saying hosted by: someone)
send to email: stunhp@gmail.com
i will  check out the servers and remove them if they dont work

**Comment by Stun3** [comment]
Also 2: 
thanks to @filmdestruction for keeping the old speedtest alive
much much thanks to @veso266 for finding the code and posting it to github

**Comment by Stun3** [comment]
![Screenshot 2024-05-31 205252](https://github.com/veso266/Flash-Speedtest.net-by-Ookla/assets/162827122/0568fae4-ebba-4e46-b1df-4972b9cc8585)
It's running very well
there a few things that needs fixing
-automatic IP address and ISP detection
-some of the servers work on the new speedtest but not on this one

**Comment by veso266** [comment]
I like ur website, you even made ads (maybe its better to dont use real ones so okkla wouldnt think u make money and complain)

Regarding ur automatic isp and ip detection
Sadly github only allows client code

But u could make a network request from javascript to here: https://ipinfo.io/ and get ip and isp that way, then once u get that load speedtest swf

**Comment by Stun3** [comment]
I didn't add the ads, I downloaded all of your code on github, took The last website\speedtest\c.speedtest.net ,edited the index.html  so that the flash player isn't buggy, edited the server list because GitHub pages was having problems with it being normal. However, using ublock origin on the firefox portable  made the speedtest website  look kinda empty without ads.

Regarding  the  isp and ip detection, well I can't code  or anything so I don't know how  to make  that :D
But maybe you  can help me with that

Me making this website was just me  struggling for a few days, but in the end I made  it working.

**Comment by Stun3** [comment]
Also, any fix for this? Sometimes it works and sometimes it doesn't (it's the Settings, Login, Create account options up in speedtest)
![Screenshot 2024-06-01 082141](https://github.com/veso266/Flash-Speedtest.net-by-Ookla/assets/162827122/86b191ea-a105-487e-9c7c-10375ca12609)

**Comment by veso266** [comment]
Github doesnt know what to do with php files
So just rename them to html (since there is no php code in them anyway) and edit the href html tags to point to them

Coding is fun, try it and you will enjoy it
You can try this app to learn javascript: https://play.google.com/store/apps/details?id=com.codeliber.js

And one for html: https://play.google.com/store/apps/details?id=com.codeliber.html

**Comment by Stun3** [comment]
Sorry, I really don't know  what you mean by editing the href html tags, but I will try something, maybe it will work
I always have my website backed up with the same config so in case I mess it up I can recover

**Comment by Stun3** [comment]
lol i totally forgot about this, but it still seems to be working.
I'm glad ookla doesnt have any problems with this, maybe they don't even know.....

**Comment by veso266** [comment]
I sure hope they dont know or even care 😄

**Comment by Stun3** [comment]
even if they take it down, i still have all the code stored, i would only need to change the license key of speedtest and reupload it :D

**Comment by Stun3** [comment]
@veso266 there is a little problem now that the website works but i cant speedtest even on working servers, i think its because the speedtest checks the server's crossdomain.xml file and doesnt find my website on it do it doesnt allow it to speedtest, do you think its fixable?

**Comment by veso266** [comment]
hmm, what does debug say, does it tell u anything (if u write debug in speedtest main window, you should see a log like window that tells u thinks like how long the license lasts, if its valid, etc)

**Comment by Stun3** [comment]
![Image](https://github.com/user-attachments/assets/81ff0d8c-ef85-4679-b5e9-3ebe03c56ebf)
this is what the debug window says

**Comment by Stun3** [comment]
@veso266 do you think there is any fix for that? some servers do work but some don't even tho they're still up (the servers that work don't get requested by speedtest for the crossdomain.xml)

**Comment by veso266** [comment]
Ugh so sorry, totally missed this one

Not sure what to do to be honest, hope u fix it

**Comment by Stun3** [comment]
i forgot completely about this lol, i dont even try to fix it much anymore if someone wanted to they can
the issues are:
some servers dont work, your ip address and ISP doesnt show, the final result screen is a bit broken and the sharing links dont work....
but the core of it mostly works, just not with ruffle... i guess thats whats important (i removed ads too just in case)

**Comment by veso266** [comment]
yea, the onlything I have no fix for is some servers not working (I guess this naturally happens as they are not hosted by you, and who knows how much will the protocol change over time) and how would I be able to save the map without talking to much hdd space (map does work for now, but since its hosted by ookla, u never know when it will finally break)

ip address and ISP doesnt show its an easy fix (just have to generate servers.php dinamicly (not statically)
as for sharing links and result screen being broken, everything to fix theese is here in this repo (I made sure to save as much of this stuff as I could, when I still had a chance)

But I hope someone else will someday do theese fixes, cuz sadly my time is very limited nowadays...

**Comment by Stun3** [comment]
ohh so the map tiles are from ookla? they dont look much complicated tho... id say maybe like 1-3 gb?
the servers not working is interesting.... they are still up and all the required files are on them....

**Comment by veso266** [comment]
Here is how the map tiles work and how could u save them: https://github.com/AliFlux/MapTilesDownloader/issues/15

if u have time u can try it out
once u download them, u can then just change a DNS A entry (or edit your hosts file) for tiles.cdnst.net to point to ur own server and arange ur map tiles according to the structure issue tells u

**Comment by Stun3** [comment]
i'm gonna try this i have a 1tb hdd i use for storing stuff like this... so ima try it

**Comment by Stun3** [comment]
nvm i wont make it work.. the map tiles downloader app is broken... its just a blank map screen, apparently some mapbox token is broken and to make an account on there to get my own token it needs a credit card (which i dont have)

**Comment by veso266** [comment]
thats wierd, because if I understand this right, mapbox is just a map provider (like u need api token to get their maps) so u can select an area

but what u want to do is download free maps hosted by ookla, this has nothing to do with mapbox

you can also try other downloaders like this: https://github.com/riatelab/maptiles (this is written in a language called R, a nice IDE for R language is RStudio)

they have a nice exmple like this
```
# define the tile server parameters
osmpos <- create_provider(
  name = "CARTO.POSITRON",
  url = "https://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}{r}.png",
  sub = c("a", "b", "c", "d"),
  citation = "© OpenStreetMap contributors © CARTO "
)
# dowload tiles and compose raster (SpatRaster)
nc_osmpos <- get_tiles(
  x = nc, provider = osmpos, crop = TRUE,
  cachedir = tempdir(), verbose = TRUE, retina = TRUE
)
#> Zoom: 7
#> Source(s): © OpenStreetMap contributors © CARTO 
#> Cache directory: /tmp/RtmpShod4e/CARTO.POSITRON
#> 8 tiles
#> The resulting raster uses high resolution tiles.
# display map
plot_tiles(nc_osmpos)
# display credits
mtext(text = get_credit(osmpos), side = 1, line = -1, adj = .99)
```
principle is the same as with the first tool (it just that the first tool is maybe more user friendly)

**Comment by Stun3** [comment]
i asked chatgpt it spit out some python script or whatever (i'm not into coding or stuff at all) 
i made the file, ran with some command chatgpt gave me and its downloading now for around an hour, i chose zoom from 0 to 7
edit: it ended up to be 23.3 MB

**Comment by veso266** [comment]
nice, so if u have it all, you can try pointing your speedtest to it and see if it looks ok

**Comment by Stun3** [comment]
idk how tho hahah its files on my hdd in a folder called "tiles", it made subfolders from 0 to 6, and inside are more subfolders, but if i dig in deep enough it gives me the tile files

<img width="1127" height="619" alt="Image" src="https://github.com/user-attachments/assets/ac3c083a-733b-435a-bb70-fe3794430e80" />

**Comment by veso266** [comment]
do u have a webserver? (I see u use windows, so u can try [wamp](https://www.wampserver.com/en/))

then u point your computer ip (with hosts file) to this: tiles.cdnst.net

and create subfolder base in it, and put your tiles in it
so ur folder structure looks something like this

wamp64->www->base->6->34->24.png

then when u hit this url with ur browser: http://tiles.cdnst.net/base/6/34/24.png
you should see the right map tile

(to make sure its realy your hosted tiles, not ooklas, you can disable internet just to check)

BTW: I can't believe that tiles would end up taking so little space, at first I thought it will be many GB of space (since this are pictures)

**Comment by Stun3** [comment]
ok ill try, going to the same directory as their real website, which is G:\tiles\tiles\6\34\24.png is the same image as their real website

<img width="4480" height="1440" alt="Image" src="https://github.com/user-attachments/assets/cdcb2bb9-1744-4285-b148-1fb29c774d24" />

**Comment by Stun3** [comment]
<img width="1129" height="621" alt="Image" src="https://github.com/user-attachments/assets/69306cd0-b7a3-4fbf-a1d8-f06c4ac5fcf4" />
okay i've set it up like this

<img width="352" height="480" alt="Image" src="https://github.com/user-attachments/assets/da34822d-efd1-4c8a-b217-cd3a8cdbe71d" />
and yes the tiles are somehow only 23.3mb, i expected around 1-3 gb

**Comment by Stun3** [comment]
<img width="2559" height="1439" alt="Image" src="https://github.com/user-attachments/assets/dbe83d23-27bc-4f34-b15a-f809f3cfaf3e" /> 
okay i unplugged internet, did my hosts file to redirect "192.168.12.52 to tiles.cdnst.net" and it worked, i tried files that i havent loaded from their real website yet (in case chrome uses some caching) and it worked

**Comment by veso266** [comment]
nice, now if everything is well the tiles should also work from speedtest client
you can do 2 tests (while internet is unplugged)
1. serve your tiles, if they load, yey
2. don't serve your tiles (you should see a blank windows without map)

**Comment by Stun3** [comment]
how do i do that if my speedtest is hosted on github.... i would have to host speedtest AND the tiles on my pc for it to work, right?

**Comment by veso266** [comment]
yep (just copy your github website to some local directory in wamp (edit the license (I made a keygen more preaty)) and enjoy

BTW: Someone features this repo in an article: https://speedtesttest.zomdir.com/speedtests/ookla/ooklas-flash-based-speed-test.html (its a nice fealing that something u did many years ago people apriciate, although, I do hope this doesn't become to famous 😄 )

**Comment by Stun3** [comment]
i might try it hahaha if you improved the keygen good, the old one was extremely stupid (but at least it worked, good job for that)
also yeah an edit 1 hr later.. lol. you did a good thing by backing it up, i loved this old speedtest, bad to see ookla got rid of it

**Comment by veso266** [comment]
Did u managed to try it?
Could u share your MAP tiles and ur downloader script (so I can put it here for the future)

**Comment by Stun3** [comment]
[tiles.zip](https://github.com/user-attachments/files/22143178/tiles.zip)
heres the zip it has the tiles and the script from chatgpt in it
i didnt try it together with speedtest yet school started and i dont have much time for now... ill try thru the weekend
